### ISSUE TITLE
Accessibility name evaluation for QW-ACT-R11 seems to fail with visually hidden text in a button

### ISSUE THREAD

**Issue by BogdanCerovac** [qualweb/act-rules#21]
Clean install of  latest NPM `"@qualweb/core": "^0.7.34"`, JSON reports "version": "3.0.0",

Just tested a simple HTML button with screen-reader only CSS class and it failed the "Button has accessible name" rule (QW-ACT-R11) although I can see the accessible name when I used Chrome accessibility inspector (and Firefox).

Seems to me that it's a bug and causes WCAG SC 4.1.2 to fail (false positive):

Found example on a website, then isolated the HTML and tested only HTML - same result.

HTML used:

```
<style>
  .visually-hidden,
  .visually-hidden-focusable:not(:focus):not(:focus-within) {
    position: absolute !important;
    width: 1px !important;
    height: 1px !important;
    padding: 0 !important;
    margin: -1px !important;
    overflow: hidden !important;
    clip: rect(0, 0, 0, 0) !important;
    white-space: nowrap !important;
    border: 0 !important;
  }

  .bi-moon-fill::before {
    content: "\f494";
  }

  .bi-brightness-high-fill::before {
    content: "\f1d1";
  }

</style>
<button id="darkmode-button" class="btn btn-link-inline py-0 mx-3">
  <span class="d-none d-light-inline"><i class="bi bi-moon-fill" aria-hidden="true"></i><span class="visually-hidden">Dark</span></span>
  <span class="d-none d-dark-inline" hidden><i class="bi bi-brightness-high-fill" aria-hidden="true"></i><span class="visually-hidden">Light</span></span>
</button>
```

`<span class="visually-hidden">Dark</span>` is the correct accessible name for this button considering CSS and `hidden` attribute on the part with `<span class="visually-hidden">Light</span>`.

Extracted relevant data from report;

```
"QW-ACT-R11": {
            "name": "Button has accessible name",
            "code": "QW-ACT-R11",
            "mapping": "97a4e1",
            "description": "This rule checks that each button element has an accessible name.",
            "metadata": {
              "target": {
                "element": [
                  "button",
                  "input"
                ],
                "attributes": [
                  "role='button'"
                ]
              },
              "success-criteria": [
                {
                  "name": "4.1.2",
                  "level": "A",
                  "principle": "Robust",
                  "url": "https://www.w3.org/WAI/WCAG21/Understanding/name-role-value",
                  "url_tr": "https://www.w3.org/TR/WCAG21/#name-role-value"
                }
              ],
              "related": [],
              "url": "https://act-rules.github.io/rules/97a4e1",
              "type": [
                "ACTRule",
                "TestCase"
              ],
              "a11yReq": [],
              "passed": 0,
              "warning": 0,
              "failed": 1,
              "inapplicable": 0,
              "outcome": "failed",
              "description": "The test target doesn't have an accessible name, or it's empty ('')."
            },
            "results": [
              {
                "verdict": "failed",
                "description": "The test target doesn't have an accessible name, or it's empty ('').",
                "resultCode": "F1",
                "elements": [
                  {
                    "htmlCode": "<button id=\"darkmode-button\" class=\"btn btn-link-inline py-0 mx-3\"> <span class=\"d-none d-light-inline\"><i class=\"bi bi-moon-fill\" aria-hidden=\"true\" _selector=\"html > body:nth-child(2) > button:nth-child(1) > span:nth-child(1) > i:nth-child(1)\" _cssrules=\"true\"></i><span class=\"visually-hidden\" _selector=\"html > body:nth-child(2) > button:nth-child(1) > span:nth-child(1) > span:nth-child(2)\" _cssrules=\"true\">Dark</span></span> <span class=\"d-none d-dark-inline\" hidden=\"\"><i class=\"bi bi-brightness-high-fill\" aria-hidden=\"true\" _selector=\"html > body:nth-child(2) > button:nth-child(1) > span:nth-child(2) > i:nth-child(1)\"></i><span class=\"visually-hidden\" _selector=\"html > body:nth-child(2) > button:nth-child(1) > span:nth-child(2) > span:nth-child(2)\" _cssrules=\"true\">Light</span></span></button>",
                    "pointer": "html > body:nth-child(2) > button:nth-child(1)",
                    "accessibleName": ""
                  }
                ],
                "attributes": []
              }
            ]
          },
```

And you can use this JSfiddle to check it manually: https://jsfiddle.net/yz8eL74r/
